tendrl/node_agent/manager/manager.py

```diff
@@ -227,6 +228,29 @@ def on_pull(self, raw_data):
                     node_id=raw_data["node_id"],
                 )
             )
+        if "disks" in raw_data:
+            LOG.info("on_pull, Updating disks")
+            try:
+                self.etcd_client.delete(
+                    ("nodes/%s/Disks") % raw_data["node_id"], recursive=True)
+            except etcd.EtcdKeyNotFound as ex:
+                LOG.debug("Given key is not present in etcd . %s", ex)
+            disks = raw_data['disks']
+            if "disks" in disks:
+                for disk in disks['disks']:
+                    disk['node_id'] = raw_data['node_id']
+                    disk_obj = Disk(disk)
+                    disk_json = disk_obj.to_json_string()
```

Comment:
@shtripat yes this is the exact problem, that is the reason i am converting class object as json